### PR TITLE
gh #249 dsDisplay L3 Python update

### PR DESCRIPTION
**Pull request by Sumip98**
* Added script to run all L3 tests
* updated logic for latest ut_raft ConfigRead API change
* Added summary Log module for tests

### CODE REVIEW

File: host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_Runall_Source.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import importlib
+from pathlib import Path
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
```
Comment (thanushreevani) [inline host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_Runall_Source.py:30]: Remove this join

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def __init__(self, testName:str, qcId:str, log:logModule=None ):
```
Comment (thanushreevani) [inline host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:53]: Need to update test spec document for L3

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def __init__(self, testName:str, qcId:str, log:logModule=None ):
 
         super().__init__(testName, qcId, log)
```
Comment (thanushreevani) [inline host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:54]: Need to update test procedure doc for L3

File: host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_Runall_Source.py (hunk 1 above)
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_Runall_Source.py]: removed os.path.join

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py (hunk 2 above)
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:53]: updated the test specification document for L3

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py (hunk 3 above)
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:54]: updated the test procedure  document for L3

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:53]: Defaults removed

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py [hunk 4]
```diff
@@ -63,13 +63,15 @@ def __init__(self, testName:str, qcId:str, log:logModule=None ):
...
+
          # Create user response Class
         self.testUserResponse = utUserResponse()
 
         # Get path to device profile file
-        self.deviceProfile = os.path.join(dir_path, self.cpe.get("test").get("profile"))
+        self.moduleConfigProfileFile = os.path.join(dir_path, deviceTestSetup.get("profile"))
 
-        self.deviceDownloadPath = self.cpe.get("target_directory")
+        self.targetWorkspace = self.cpe.get("target_directory")
```
Comment (srinivasgtl) [inline host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:74]: do we need to have the target work space as well with the module name attached to it?  Probably we should follow a similar structure like in dsAudio.

File: host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_testSetup.yml [hunk 5]
```diff
@@ -5,17 +5,6 @@ dsDisplay:  # Prefix must always exist
...
-          - "<URL Path>/libut_control.so"
-          - "<URL Path>/Source_4K_Display.yaml"
-          - "<URL Path>/run.sh"
-        execute:
-          - "chmod +x <Path on dut>/dsDisplay/hal_test"
-          - "chmod +x <Path on dut>/run.sh"
-          - "ln -s /usr/lib/libds-hal.so <Path on dut>/libdshal.so"
-        streams:
       test01_VerifyDisplayConnectCallBackTest:
         <<: *defaults
```
Comment (thanushreevani) [inline host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_testSetup.yml:9]: This is also not required right

File: docs/pages/dsDisplay/ds-display_L3_Low-Level_TestSpecification.md [hunk 6]
```diff
@@ -97,6 +99,6 @@ classDiagram
   - This configuration file contains the list of requirements for tests to execute. Eg: Copying the streams, setting environment variables etc.
   - Example configuration file [dsDisplay_L3_testSetup.yml](../../../host/tests/dsDisplay_L3_Tests/dsDisplay_L3_testSetup.yml)
```
Comment (thanushreevani) [inline docs/pages/dsDisplay/ds-display_L3_Low-Level_TestSpecification.md:100]: This path needs to be modified

File: host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py (hunk 4 above)
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplayHelperClass.py:74]: updated

File: docs/pages/dsDisplay/ds-display_L3_Low-Level_TestSpecification.md (hunk 6 above)
Comment (Sumip98) [reply docs/pages/dsDisplay/ds-display_L3_Low-Level_TestSpecification.md]: modified

File: host/tests/dsClasses/dsDisplay.py [hunk 7]
```diff
@@ -250,7 +258,7 @@ def getSupportedPorts(self):
         """
         portLists= []
 
-        ports = self.deviceProfile.get("Video_Ports")
+        ports = self.moduleConfigProfile.get("Video_Ports")
         for entry in ports:
             video_port_name = dsVideoPortType(entry).name
             port_index =  0# Get the index
```
[the remark this answers is not in the document]
Comment (Sumip98) [reply host/tests/dsClasses/dsDisplay.py]: I have updated the port index in the YAML file. Indices are now read from the YAML file

File: host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_testSetup.yml (hunk 5 above)
Comment (Sumip98) [reply host/tests/L3_TestCases/dsDisplay/dsDisplay_L3_testSetup.yml]: updated

**Review by srinivasgtl — APPROVED** [review]
ship it.

**Review by thanushreevani — APPROVED** [review]

**Pull request merged**